### DISCUSSION TITLE
Reading/evaluating several files at once?

### DISCUSSION THREAD

**Discussion by braimen** [jotaen/klog#176] [Help / Question] [answered]
I would like to use one file per month, but also to get a report over a longer period. It would be cool, if klog would support not only a single file, but several. Maybe selected by a placeholder like:
`$ klog report 2022-*`
or even better with bookmarks:
`$ klog bookmarks set ./2022-*.klg @year`

Thanks for klog!

**Comment by jotaen** [comment, marked as answer]
If your shell supports glob patterns, then this should work out of the box, e.g.: `klog report **/*.klg` or `klog total 2022-*.klg`.

This doesn’t work with bookmarks, however. A bookmark must be a single file, because it’s not just meant for reading, but also for writing, e.g. via `klog start @work`. If the bookmark was a file pattern that resolved to multiple files, it would be ambiguous what should happen then.

**Reply by braimen** [reply]
Thanks a lot for the fast reply. I'll try this tomorrow. This possibility should be mentioned in the documentation! 😃
